dataclients/kubernetes/kube_test.go

```diff
@@ -614,7 +614,7 @@ func TestIngressData(t *testing.T) {
 			),
 		)},
 		map[string]string{
-			"kube_foo__qux______www_zalando_de": "https://www.zalando.de:443",
+			"kube_foo__qux__www_zalando_de____www_zalando_de": "https://www.zalando.de:443",
```

Comment:
The test is testing route id change but not the addition of `Host` predicate. 

I would be nice to modify test harness to check not only backend but rather whole route like, e.g.:
```go
map[string]string{
    "kube_foo__qux__www_zalando_de____www_zalando_de": "Host(/^www[.]zalando[.]de$/) -> setRequestHeader("Host", "www.zalando.de") -> "https://www.zalando.de:443""
```
Not sure if this is within the scope of this ticket though.

Reply:
out of scope, we have to change the tests similar to routegroup IMO

Reply:
There is `checkPrettyRoutes` that does the job.